$NetBSD: patch-configure.ac,v 1.10 2025/03/30 21:10:47 wiz Exp $

* Detect NetBSD libatomic
* test(1) portability fix
* genccode assembly for every NetBSD target, not only i386

--- configure.ac.orig
+++ configure.ac
@@ -426,6 +426,18 @@
 fi
 
 
+# check if sys/atomic.h is present.
+AC_CHECK_HEADERS([sys/atomic.h])
+case "${host}" in
+*-netbsd*)
+if test "x$ac_cv_header_sys_atomic_h" = "xyes"; then
+    CONFIG_CPPFLAGS="$CONFIG_CPPFLAGS -DU_HAVE_NETBSD_ATOMIC_OPS=1";
+fi
+;;
+*)
+;;
+esac
+
 U_ENABLE_DYLOAD=1
 enable=yes
 AC_MSG_CHECKING([whether to enable dynamic loading of plugins. Ignored if plugins disabled.])
@@ -613,7 +625,7 @@
 # Check to see if genccode can generate simple assembly.
 GENCCODE_ASSEMBLY=
 case "${host}" in
-*-linux*|*-kfreebsd*-gnu*|i*86-*-*bsd*|*-pc-gnu)
+*-linux*|*-netbsd*|*-kfreebsd*-gnu*|i*86-*-*bsd*|*-pc-gnu)
     if test "$GCC" = yes; then
         # We're using gcc, and the simple -a gcc command line works for genccode
         GENCCODE_ASSEMBLY="-a gcc"
@@ -1132,6 +1144,7 @@
 	*-*-haiku)	platform=U_HAIKU ;;
 	*-*-irix*)		platform=U_IRIX ;;
 	*-ncr-*)		platform=U_MPRAS ;;
+	*-*-sco3.2v5*)		platform=USCOOSR5 ;;
 	*) 		platform=U_UNKNOWN_PLATFORM ;;
 esac
 AC_SUBST(ICUDATA_CHAR)
